# **Predicting Perth Property Prices: A Data-Driven Journey into Smart property Investment Decisions**

*The project aims to develop a predictive model that provides estimates for property prices based on various influential factors. 
The insights from this project aim to facilitate informed decision-making in property investment and sales within the region.*

*Prepared by :* **Faris Arief Mawardi**

## **I. Introduction**

==========================================================

Dataset: perth_houses.csv

Dataset Source: [Kaggle](https://www.kaggle.com/datasets/syuzai/perth-house-prices)

This notebook is created to test the model's ability to predict property prices based on several dummy data from the features available in the constructed model.

==========================================================

# II. Import Library

In [1]:
import pickle  # To load the constructed model
import pandas as pd  # To execute several command in manipulating datas or cells

# III. Model Loading

In [2]:
# Loading the model
with open('model.pkl', 'rb') as file_1:
  model = pickle.load(file_1)

# IV. Model Inference

In [3]:
# Generating new dummy datas
data_inf = {
    "ADDRESS": ["1 Acorn Place", "9E Margaret Street"],
    "SUBURB": ["South Lake", "Midland"],
    "BEDROOMS": [3, 2],
    "BATHROOMS": [3, 2],
    "GARAGE": [1, 1],
    "LAND_AREA": [250, 246],
    "FLOOR_AREA": [120, 85],
    "BUILD_YEAR": [1990, 1996],
    "CBD_DIST": [18300, 16700],
    "NEAREST_STN": ["Cockburn Central Station", "Midland Station"],
    "NEAREST_STN_DIST": [1800, 1700],
    "DATE_SOLD": [2018, 2016],
    "POSTCODE": [6164, 6056],
    "LATITUDE": [-32.115900, -31.882163],
    "LONGITUDE": [115.842450, 116.014755],
    "NEAREST_SCH": ["LAKELAND SENIOR HIGH SCHOOL", "LA SALLE COLLEGE"],
    "NEAREST_SCH_DIST": [0.828339, 1.055564	],
    "NEAREST_SCH_RANK": [200, 53.0],
}

# Generating a new dataframe consists of the dummy datas
data_inf = pd.DataFrame(data_inf)

# Print the dataframe
data_inf

ADDRESS      SUBURB  BEDROOMS  BATHROOMS  GARAGE  LAND_AREA  \
0       1 Acorn Place  South Lake         3          3       1        250   
1  9E Margaret Street     Midland         2          2       1        246   

   FLOOR_AREA  BUILD_YEAR  CBD_DIST               NEAREST_STN  \
0         120        1990     18300  Cockburn Central Station   
1          85        1996     16700           Midland Station   

   NEAREST_STN_DIST  DATE_SOLD  POSTCODE   LATITUDE   LONGITUDE  \
0              1800       2018      6164 -32.115900  115.842450   
1              1700       2016      6056 -31.882163  116.014755   

                   NEAREST_SCH  NEAREST_SCH_DIST  NEAREST_SCH_RANK  
0  LAKELAND SENIOR HIGH SCHOOL          0.828339             200.0  
1             LA SALLE COLLEGE          1.055564              53.0

# V. Predicting The Results

In [4]:
# Predict
y_inf_pred = model.predict(data_inf)
y_inf_pred

array([427250.        , 338642.85714286])

# VI. Conclusions

**Conclusion:**

Based on the conducted inferences, the created model can predict property prices reasonably well using the provided dummy data. Below are the property price predictions generated by the constructed model:

- Prediction for the First Data: $427,250
- Prediction for the Second Data: $338,642